src/charm.py

```diff
@@ -266,6 +279,32 @@ def __init__(self, *args):
         # Action handlers
         observe(self.on.show_proxied_endpoints_action, self._on_show_proxied_endpoints)  # type: ignore
 
+    @property
+    def _forward_auth_config(self) -> ForwardAuthRequirerConfig:
+        ingress_app_names = [
+            rel.app.name  # type: ignore
+            for rel in itertools.chain(
+                self.ingress_per_appv1.relations,
+                self.ingress_per_appv2.relations,
+                self.ingress_per_unit.relations,
+                self.traefik_route.relations,
+            )
+        ]
+        return ForwardAuthRequirerConfig(ingress_app_names)
+
+    def _on_forward_auth_config_changed(self, event: AuthConfigChangedEvent):
+        if self.config["enable_experimental_forward_auth"]:
+            self.forward_auth.update_requirer_relation_data(self._forward_auth_config)
```

Comment:
Do we need to update the app_names when the provider's databag changes? I think that these should be independent actions (i.e. when the provider's databag changes, the only action needed is that traefik updates its ingress routes config).

I think that traefik should update the requirer databag when the relation is created and then on `_process_status_and_configurations`